src/tools/rbd_mirror/BaseRequest.h

```diff
@@ -0,0 +1,39 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_BASE_REQUEST_H
+#define CEPH_RBD_MIRROR_BASE_REQUEST_H
+
+#include "common/RefCountedObj.h"
+
+namespace rbd {
+namespace mirror {
+
+class BaseRequest : public RefCountedObject {
+public:
+  BaseRequest(CephContext *cct = nullptr, Context *on_finish = nullptr)
+    : RefCountedObject(cct, 1), m_cct(cct), m_on_finish(on_finish) {
+  }
+
+  virtual void send() = 0;
+  virtual void cancel() {}
+
+protected:
+  CephContext *m_cct;
+  Context *m_on_finish;
+
+  void finish(int r) {
+    if (m_cct) {
+      lsubdout(m_cct, rbd_mirror, 20) << "finish: r=" << r << dendl;
```

Comment:
Minor: include the class name in the output to trace where "finish: r=XYZ" comes from in the log